packages/dev/core/src/Engines/renderTargetWrapper.ts

```diff
@@ -289,6 +294,8 @@ export class RenderTargetWrapper {
             renderTarget._depthStencilTexture = this._depthStencilTexture;
             this._depthStencilTexture.incrementReferences();
         }
+
+        this._size = renderTarget._size;

```

Comment:
Is it not mandatory to be the same ?

Reply:
size is not updated automatically when sharing textures amongst framebuffer. Since depthbuffer has to be always the same size as other textures bound in the FB, I found it was a good place to update the size

Reply:
yes what I am saying is that if the size are different it will crash isnt it ?

So should it not be a check rather than an assignment ?

Reply:
In that case we have to provide a function to replace textures + depth buffer at once, because for now `setTexture` and `_shareDepth` are two different calls, making a temporary discrepancy in the size

Reply:
I understand now!

I think we should update `setTexture` instead, to set the size according to the texture passed in.